gadgets/cses-status/src/lib.rs

```diff
@@ -1,24 +1,73 @@
 use std::sync::{Arc, Mutex};
 
 use cses_query::{Problem, ProblemStatus};
-use egui::{Color32, CursorIcon, RichText, Sense};
+use egui::{Color32, CursorIcon, RichText, Sense, TextEdit};
+use serde::{Deserialize, Serialize};
 use utils::{Gadget, GadgetFactory, MutexExt};
 
 mod cses_query;
 
 pub struct CSESStatusGadget {
-    problem_data: Arc<Mutex<Vec<Problem>>>,
+    problem_data: ProblemData,
+    network_runtime: utils::NetworkRuntime,
+    ask_for_id: bool,
+    input: String,
 }
 
+type ProblemData = Arc<Mutex<Vec<Problem>>>;
+
 pub struct CSESStatusGadgetFactory;
 
+#[derive(Serialize, Deserialize)]
+struct CSESStatusConfig {
+    #[serde(rename = "user-id")]
+    user_id: u64,
+}
+
+impl CSESStatusGadget {
+    fn ask_for_id(&mut self, ui: &mut egui::Ui, ctx: egui::Context) {
```

Comment:
Is there a reason for passing a `Context` instead of calling `Ui::ctx()`? (https://docs.rs/egui/latest/egui/struct.Ui.html#method.ctx)

Reply:
Was messing with `Ui::columns` a bit and getting the context, was a bit confusing for me, I'll change that